workflows/parameterizing-a-workflow.md

```diff
@@ -0,0 +1,33 @@
+# Workflow Parameters
+
+Oftentimes, arguments to workflows might not be known until runtime. A workflow might even need to be run multiple times with different parameters.
+
+We allow you to parameterize your workflows with a special type of artifact called a Parameter artifact. You can create such an artifact through the Aqueduct 
+client with `client.create_param(<name>, default=<default_val>)`. The returned artifact can then be consumed by operators as input, just like any other
+artifact (eg. `predict_op(param_artifact)`). The parameter can also be used in SQL query strings with the `{{<param_name>}}` syntax, where the double-bracket pattern 
+is expanded automatically into the parameter value by the system.
+
+Every parameter must have a default value. This is so that when a workflow is run without explicitly defined parameters, it always has a value it can use.
+The value of a parameter can be any type, as long as Aqueduct can convert it into a json string with `JSON.dumps()`. For parameters used in SQL queries, their
+value must be a string type.
```

Comment:
Do we not an integer/bool/float? Or do you just mean it has to be a primitive type?

Reply:
for SQL queries, it has to be a string right now (eg. "123" not 123).

Reply:
do we do type inference? or do we only support comparisons against string columns?

Reply:
we don't do any type inference right now. It's just a straightforward string replacement, so for string columns the user actually has to remember to wrap the parameter value in quotes, eg: `select * from hotel_reviews where col_name={{param}}`, the value of param need to be "'<col_name>'" in order to work. I think the user should be able to compare against non-string columns with right parameter string.

Reply:
Okay, gotcha! So if I wanted to compare against a string column, the user would have to include extra quotes in their string itself, right? Because otherwise we drop the quotes -- so if I set `param` to `"hello"`, I would get `WHERE col = hello`, whereas if I set `param` to `"\'hello\'"`, I would get `WHERE col = "hello"`.

Reply:
Oh, I see, we're expecting the user to put it into their SQL query! Okay, sorry for the pile of questions. It all makes sense now.

Let's just add a note here saying that if you use strings, you have to insert quotes yourself, and then let's merge this.